src/main/java/de/zalando/aruha/nakadi/service/subscription/KafkaClient.java

```diff
@@ -0,0 +1,38 @@
+package de.zalando.aruha.nakadi.service.subscription;
+
+import de.zalando.aruha.nakadi.domain.Cursor;
+import de.zalando.aruha.nakadi.domain.Subscription;
+import de.zalando.aruha.nakadi.exceptions.NakadiException;
+import de.zalando.aruha.nakadi.repository.kafka.KafkaTopicRepository;
+import de.zalando.aruha.nakadi.service.subscription.model.Partition;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class KafkaClient {
+    private final Subscription subscription;
+    private final KafkaTopicRepository topicRepository;
+
+    KafkaClient(final Subscription subscription, final KafkaTopicRepository topicRepository) {
+        this.subscription = subscription;
```

Comment:
Why have you decided to have this class stateful? (as well as CuratorZkSubscriptionClient)
It's not possible to have it as Spring bean as it has to be created for each subscription, and it's difficult to write tests for classes that use this class (additional factory is needed to allow mocking this class in test code)